docs/Permissions/Onchain-Permissioning.md

```diff
@@ -0,0 +1,191 @@
+to a static address. 
+
+* Rules - stores the node whitelist and list of admin accounts. 
+
+## Pre-requisites 
+
+For nodes that are going to set up or maintain the nodes whitelist: 
+
+* [NodeJS](https://nodejs.org/en/) v8.9.4 or later 
+
+* [Truffle](https://truffleframework.com/docs/truffle/getting-started/installation)
+
+## Add Ingress Contract to Genesis File
+
+Add the Ingress contract to the genesis file for your network by copying it from [`genesis.json`](https://github.com/PegaSysEng/permissioning-smart-contracts/blob/master/genesis.json) 
+in the [`permissioning-smart-contracts` repository](https://github.com/PegaSysEng/permissioning-smart-contracts): 
+   
+```json
+"0x0000000000000000000000000000000000009999": {
+      "comment": "Ingress smart contract",
+      "balance": "0",
+      "code": <stripped>,
+      "storage": {
+         <stripped>
+      }
+}
+```
+
+!!! important 
+    Ensure your genesis file includes a sufficient milestone to support the permissioning contracts. 
+    For example, `constantinopleFixBlock`. 
+    
+## Onchain Permissioning Setup
+   
+1. Start your Pantheon node with contract permissioning options: 
+
+    * [--permissions-nodes-contract-enabled](../Reference/Pantheon-CLI-Syntax.md#permissions-nodes-contract-enabled)
+      to enable onhcain permissioning
+
+    * [--permissions-nodes-contract-address](../Reference/Pantheon-CLI-Syntax.md#permissions-nodes-contract-address)
+      set to the address of the Ingress contract in the genesis file (`"0x0000000000000000000000000000000000009999"`)
+
+1. Create the following environment variables and set to the specified values: 
+
+    * `PANTHEON_NODE_PERM_ACCOUNT` - address of account used to interact with the permissioning contracts. 
+
+    * `PANTHEON_NODE_PERM_KEY` - private key of the account used to interact with the permissioning contracts.
+
+    * `INGRESS_CONTRACT_ADDRESS` - address of the Ingress contract in the genesis file.  
+
+    * `PANTHEON_NODE_PERM_ENDPOINT` - required only if your node is not using the default JSON-RPC host and port (`http://127.0.0.1:8545`). 
+       Set to JSON-RPC host and port. 
+
+    !!! note
+        If your network is not configured as a [free gas network](../Configuring-Pantheon/FreeGas.md), the account used to 
+        interact with the permissioning contracts must have a balance. 
+
+1. Clone the `permissioning-smart-contracts` repository: 
+
+    ```bash 
+    git clone https://github.com/PegaSysEng/permissioning-smart-contracts.git
+    ```
+
+1. Change into the `permissioning-smart-contracts` directory and run: 
+
+    ```bash
+    npm install
+    ```
+
+## Deploy Rules Contract 
+
+In the `permissioning-smart-contracts` directory, deploy the Rules contract: 
+
+```bash
+truffle migrate --reset
+```
+
+The Rules contract is deployed and the Ingress contract updated with the name and version of the Rules contract. 
```

Comment:
The migration logs the address of the Rules contract.

Reply:
Fixed